include/z64audio.h

```diff
@@ -0,0 +1,495 @@
+ */
+
+struct Note;
+struct NotePool;
+struct SequenceChannel;
+struct SequenceChannelLayer;
+
+typedef struct AudioListItem {
+    // A node in a circularly linked list. Each node is either a head or an item:
+    // - Items can be either detached (prev = NULL), or attached to a list.
+    //   'value' points to something of interest.
+    // - List heads are always attached; if a list is empty, its head points
+    //   to itself. 'count' contains the size of the list.
+    // If the list holds notes, 'pool' points back to the pool where it lives.
+    // Otherwise, that member is NULL.
+    /*0x00*/ struct AudioListItem* prev;
+    /*0x04*/ struct AudioListItem* next;
+    /*0x08*/ union {
+                 void* value; // either Note* or SequenceChannelLayer*
+                 s32 count;
+             } u;
+    /*0x0C*/struct NotePool* pool;
+} AudioListItem; // size = 0x10
+
+// UNCHANGED
+typedef struct NotePool {
+    AudioListItem disabled;
+    AudioListItem decaying;
+    AudioListItem releasing;
+    AudioListItem active;
+} NotePool;
+
+// Pitch sliding by up to one octave in the positive direction. Negative
+// direction is "supported" by setting extent to be negative. The code
+// exterpolates exponentially in the wrong direction in that case, but that
+// doesn't prevent seqplayer from doing it, AFAICT.
+typedef struct {
+    u8 mode; // bit 0x80 denotes something; the rest are an index 0-5
+    f32 cur;
+    f32 speed;
+    // f32 extent;
+    // struct Note *note; // Changed from extent to note? Shortened Struct maybe?
+} Portamento; // size = 0x10
+
+typedef struct {
+    s16 delay;
+    s16 arg;
+} AdsrEnvelope; // size = 0x4
+
+typedef struct {
+    u32 start;
+    u32 end;
+    u32 count;
+    u32 pad;
+    s16 state[16]; // only exists if count != 0. 8-byte aligned
+} AdpcmLoop;
+
+typedef struct {
+    s32 order;
+    s32 npredictors;
+    s16 book[1]; // size 8 * order * npredictors. 8-byte aligned
+} AdpcmBook;
+
+typedef struct {
+    u8 unused;
+    u8 loaded;
+    u8* sampleAddr;
+    AdpcmLoop* loop;
+    struct AdpcmBook* book;
+    u32 sampleSize; // never read. either 0 or 1 mod 9, depending on padding
+} AudioBankSample;
+
+typedef struct {
+    AudioBankSample* sample;
+    f32 tuning; // frequency scale factor
+} AudioBankSound;              // size = 0x8
+
+typedef struct {
+    /*0x00*/ u8 loaded;
+    /*0x01*/ u8 normalRangeLo;
+    /*0x02*/ u8 normalRangeHi;
+    /*0x03*/ u8 releaseRate;
+    /*0x04*/ AdsrEnvelope* envelope;
+    /*0x08*/ AudioBankSound lowNotesSound;
+    /*0x10*/ AudioBankSound normalNotesSound;
+    /*0x18*/ AudioBankSound highNotesSound;
+} Instrument; // size = 0x20
+
+typedef struct {
+    u32 drum0;
+    u32 drum4;
+    // u8 releaseRate;
+    // u8 pan;
+    // u8 loaded;
+    // AudioBankSound sound;
+    // AdsrEnvelope *envelope;
+} Drum; // OOT size = 0x8, TODO figure out what is still used.
+
+typedef struct {
+    u32 ootNew;
```

Comment:
Personally I'm not a fan of the naming scheme here, should keep up with our unknown naming scheme, comments can be made where differences apply from SM64 
```suggestion
    u32 unk_00; // new in OOT from sm64
    u32 unk_04; // new in OOT from sm64
```